src/il/int_posix.c

```diff
@@ -490,6 +490,7 @@ IOF_PUBLIC int iof_open(const char *pathname, int flags, ...)
 		fd = __real_open(pathname, flags, mode);
 	} else {
 		fd =  __real_open(pathname, flags);
+		mode = 0;
```

Comment:
nice catch :-)

Reply:
There are clearly some different compiler flags in use by DAOS, I don't know how they caught this when we didn't.  Mind you, logically it has no effect as mode is only printed inside a conditional that should only be true if mode has actually been set.